### ISSUE TITLE
`rules-of-hooks` thinks that `userToken` is a hook because it starts with `use`

### ISSUE THREAD

**Issue by lucacasonato** [denoland/deno_lint#1393]
### Lint Name

`rules-of-hooks`

### Code Snippet

```ts
function userHook() {}

function doAThing() {
  userHook()
}
```

### Expected Result

No lint.

### Actual Result

```
error[rules-of-hooks]: Hook function called outside of a Component
 --> /home/lucacasonato/projects/github.com/denoland/deno/test.tsx:4:3
  | 
4 |   userHook()
  |   ^^^^^^^^^^
  = hint: Ensure that hooks are only called inside Components or custom hook functions

  docs: https://lint.deno.land/rules/rules-of-hooks


Found 1 problem
Checked 1 file
```

### Version

deno 2.1.7+e475749
